progress/internal/linear-progress.ts

```diff
@@ -30,14 +31,14 @@ export class LinearProgress extends Progress {
     };
     const dotStyles = {
       transform: `scaleX(${
-        (this.indeterminate ? 1 : this.buffer / this.max) * 100
+        (this.indeterminate || this.buffer < 0 ? 1 : this.buffer / this.max) * 100
```

Comment:
Does `this.buffer <= 0` handle the null case correctly?

When `null`, it'll coerce to `0`. If a buffer is `0`, it won't display anyway.

Reply:
@asyncLiz unfortunately not. `dotStyles` is evaluated to `scaleX(0%)` but `hideDots` is still evaluated to `false` so the whole buffer shows up after removing it:

![image](https://github.com/material-components/material-web/assets/6388546/7bd2d338-9847-4c50-87ce-5b5e62caed04)

![image](https://github.com/material-components/material-web/assets/6388546/40a0ee0e-546d-4588-ac78-b49d9c5054d7)

I am making another commit to fix it soon.

Reply:
New commit should now correctly handle removing `buffer` attribute. I am not sure if it's better to set its type to `number | null` because it reflects reality better and possibly saves us from some unintended bug in the future. Nevertheless you are more aware of the whole build process than me.